src/contracts/pods/EigenPod.sol

```diff
@@ -576,151 +554,150 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
-            verifiedWithdrawal.amountToSend =
-                uint256(withdrawalAmountGwei - MAX_RESTAKED_BALANCE_GWEI_PER_VALIDATOR) *
-                uint256(GWEI_TO_WEI);
-            // and the extra execution layer ETH in the contract is MAX_RESTAKED_BALANCE_GWEI_PER_VALIDATOR, which must be withdrawn through EigenLayer's normal withdrawal process
-            withdrawableRestakedExecutionLayerGwei += MAX_RESTAKED_BALANCE_GWEI_PER_VALIDATOR;
-            withdrawalAmountWei = MAX_RESTAKED_BALANCE_GWEI_PER_VALIDATOR * GWEI_TO_WEI;
+            amountToQueueGwei = MAX_RESTAKED_BALANCE_GWEI_PER_VALIDATOR;
         } else {
-            // otherwise, just use the full withdrawal amount to continue to "back" the podOwner's remaining shares in EigenLayer
-            // (i.e. none is instantly withdrawable)
-            withdrawableRestakedExecutionLayerGwei += withdrawalAmountGwei;
-            withdrawalAmountWei = withdrawalAmountGwei * GWEI_TO_WEI;
-        }
-        // if the amount being withdrawn is not equal to the current accounted for validator balance, an update must be made
-        if (currentValidatorRestakedBalanceWei != withdrawalAmountWei) {
-            verifiedWithdrawal.sharesDelta = _calculateSharesDelta({
-                newAmountWei: withdrawalAmountWei,
-                currentAmountWei: currentValidatorRestakedBalanceWei
-            });
+            amountToQueueGwei = withdrawalAmountGwei;
         }
 
+        /**
+         * If the withdrawal is for more than the max per-validator balance, we mark 
+         * the max as "withdrawable" via the queue, and withdraw the excess immediately
+         */
+
+        VerifiedWithdrawal memory verifiedWithdrawal;
+        verifiedWithdrawal.amountToSendGwei = uint256(withdrawalAmountGwei - amountToQueueGwei);
+        withdrawableRestakedExecutionLayerGwei += amountToQueueGwei;
+        
+        /**
+         * Next, calculate the change in number of shares this validator is "backing":
+         * - Anything that needs to go through the withdrawal queue IS backed
+         * - Anything immediately withdrawn IS NOT backed
+         *
+         * This means that this validator is currently backing `amountToQueueGwei` shares.
+         */
+
+        verifiedWithdrawal.sharesDeltaGwei = _calculateSharesDelta({
+            newAmountGwei: amountToQueueGwei,
+            previousAmountGwei: validatorInfo.restakedBalanceGwei
+        });
+
+        /**
+         * Finally, the validator is fully withdrawn. Update their status and place in state:
+         */
 
-        // now that the validator has been proven to be withdrawn, we can set their restaked balance to 0
         validatorInfo.restakedBalanceGwei = 0;
         validatorInfo.status = VALIDATOR_STATUS.WITHDRAWN;
-        validatorInfo.mostRecentBalanceUpdateTimestamp = withdrawalHappenedTimestamp;
-
+        validatorInfo.mostRecentBalanceUpdateTimestamp = withdrawalTimestamp;
```

Comment:
going to point out that elsewhere we set this field equal to the `oracleTimestamp`, whereas here we are setting it to when the withdrawal actually occurred
AFAICT there's no actual issue present here, but wanted to note the inconsistency in case there is one

Reply:
yup, thats because its the last balance update made becuase its a withdrawal.  With the other proofs, the oracleTimestamp is the timstamp of the update, whereas here its timestamp of the withdrawal which may be far in the past

Reply:
that's a great thing to mention. I was thinking that this line might not be quite right.

Imagine a scenario where you manage to hit `_processFullWithdrawal` with multiple withdrawals, but you purposefully prove them out of order. Then the validator's `mostRecentBalanceUpdateTimestamp` isn't really accurate.

WDYT?

Reply:
great point. i think this is not critical, since the logic around `mostRecentBalanceUpdateTimestamp` only matters when the validator is not active but still we should probably only allow it to increase

Reply:
> WDYT?

Despite this appearing to have no security implications, I would prefer to not leave known inconsistencies in variable usage in.
My recommendation is passing the `oracleTimestamp` into this function and using it here. **However**, in the edge case of proving multiple withdrawals for the same validator at the same time, I think this would lead to a revert still -- would there be an issue with making these checks against the oracleTimestamp  (`validatorInfo.mostRecentBalanceUpdateTimestamp < oracleTimestamp`) _non-strict_ instead of strict? Because then we'd also solve this pesky edge case.